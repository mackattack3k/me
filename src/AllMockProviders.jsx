import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import defaultTheme from './features/theme/defaultTheme';
import GithubMockProvider from './testproviders/GithubMockProvider';

const AllMockProviders = ({ children }) => (
  <ThemeProvider theme={defaultTheme.dark}>
    <GithubMockProvider>{children}</GithubMockProvider>
  </ThemeProvider>
);

AllMockProviders.propTypes = {
  children: PropTypes.node.isRequired
};

export default AllMockProviders;
