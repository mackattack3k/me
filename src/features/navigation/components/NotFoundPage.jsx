import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import H1 from '../../text/components/H1';

const Page = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: ${props => props.theme.secondary.dark};
  text-align: center;
`;
const SubTitle = styled.div`
  color: ${props => props.theme.secondary.dark};
  font-size: 18px;
`;

const NotFoundPage = () => {
  const { t } = useTranslation();
  return (
    <Page>
      <H1>{t('not_found.title')}</H1>
      <SubTitle>{t('not_found.subtitle')}</SubTitle>
    </Page>
  );
};

export default NotFoundPage;
