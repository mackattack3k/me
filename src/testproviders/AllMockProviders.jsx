import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../features/theme/defaultTheme';
import { ToastProvider } from '../features/toast/components/ToastProvider';
import GithubMockProvider from './GithubMockProvider';

const AllMockProviders = ({ children }) => (
  <ThemeProvider theme={defaultTheme.dark}>
    <ToastProvider>
      <GithubMockProvider>{children}</GithubMockProvider>
    </ToastProvider>
  </ThemeProvider>
);

AllMockProviders.propTypes = {
  children: PropTypes.node.isRequired
};

export default AllMockProviders;
