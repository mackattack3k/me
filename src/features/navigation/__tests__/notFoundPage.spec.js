import { render } from '@testing-library/react';
import React from 'react';
import AllMockProviders from '../../../testproviders/AllMockProviders';
import NotFoundPage from '../components/NotFoundPage';

describe('not found page', () => {
  it('should render with snapshot', () => {
    const pageNotFound = render(
      <AllMockProviders>
        <NotFoundPage />
      </AllMockProviders>
    );
    expect(pageNotFound).toMatchSnapshot();
  });
});
