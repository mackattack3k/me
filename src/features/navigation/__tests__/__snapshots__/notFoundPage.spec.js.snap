// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not found page should render with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c3 {
  position: absolute;
  bottom: 0;
  left: 0;
}

.c1 {
  font-family: 'Proza Libre',sans-serif;
}

.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #c79a00;
  text-align: center;
}

.c2 {
  color: #c79a00;
  font-size: 18px;
}

<body>
    <div>
      <div
        class="c0"
      >
        <h1
          class="c1"
        >
          404 Page not found
        </h1>
        <div
          class="c2"
        >
          Are you sure you are where you want to be? ðŸ¤”
        </div>
      </div>
      <div
        class="c3"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bxivhb cQzMmm"
    >
      <h1
        class="sc-htpNat bDALxL"
      >
        404 Page not found
      </h1>
      <div
        class="sc-ifAKCX eIgqsz"
      >
        Are you sure you are where you want to be? ðŸ¤”
      </div>
    </div>
    <div
      class="sc-bwzfXH ftCJGa"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
